### PR TITLE
Add sqlite and logging and more...

### PR DESCRIPTION
**Pull request by callunity**

### CODE REVIEW

File: components/fileqcs/fileQcsController.js [hunk 1]
```diff
@@ -1,220 +1,519 @@
 'use strict';
 
-const pgp = require('pg-promise')();
-const db = pgp(process.env.DB_CONNECTION);
+const pgp = require('pg-promise')({});
+const pg = pgp(process.env.DB_CONNECTION);
+const sqlite3 = require('sqlite3').verbose(); // TODO: remove `verbose()` in production
```
Comment (apmasell) [inline components/fileqcs/fileQcsController.js]:
```
const basesqlite3 = require('sqlite3);
const sqlite3 = ((process.env.DEBUG || 'false') === 'true') ? basesqlite3.verbose() : basesqlite3;
```

File: components/fileqcs/fileQcsController.js [hunk 2]
```diff
@@ -1,220 +1,519 @@
...
+        logger.error(err);
+        reject(generateError(500, 'Error retrieving record'));
+      });
+  });
+}
+
+/** success returns an array of File Provenance results filtered by the given project */
+function getFprResultsByProject (project) {
+  return new Promise((resolve, reject) => {
+    fpr.all('SELECT * FROM fpr WHERE project = ? ORDER BY fileswid ASC', [project], (err, data) => {
```
Comment (apmasell) [inline components/fileqcs/fileQcsController.js]: Some of these `?` and some are `$1` and some are `${x}`. Is there a reason?

Comment (callunity) [reply components/fileqcs/fileQcsController.js]: `?` is for SQLite, and the `$` ones are Postgres. I've used `${x}` where there are three or more arguments (and in these cases, the arguments are coming into the function as an object already), whereas `$1` is used when working with just a couple of parameters. If you think it's needlessly complicated, I'll use the `${x}` syntax for all Postgres statements and create objects for the 1-2 parameter statements.

Comment (callunity) [reply components/fileqcs/fileQcsController.js]: I've added a comment to the file which indicates the conventions I've used for parameterization.

**Pull request merged**